tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/server/LVQTest.java

```diff
@@ -167,11 +167,13 @@ public void testMultipleMessages() throws Exception {
       ClientMessage m = consumer.receive(1000);
       Assert.assertNotNull(m);
       m.acknowledge();
-      Assert.assertEquals(m.getBodyBuffer().readString(), "m3");
+      String val = m.getBodyBuffer().readString();
+      Assert.assertTrue("1 or 3 =? " + val, "m1".equals(val) || "m3".equals(val));
```

Comment:
This doesnt seem like a desirable change for an LVQ test. It would then be about as well deleted it for all the good it would really be doing in verifying the 'LV' bit.

It currently creates the consumer before sending so it seems rather pot-luck what it will see. The next test creates it after, and also checks the count as you said. Or it could use a transacted producer for the 'second round' of sends. Something synchronous between send and check, basically.

Reply:
I think the crux is that the broker decides what is the LV, and it is non deterministic in this case.

Reply:
It is currently but per your comment and other tests it could probably be made deterministic. Instead, with this change the test wouldnt really be verifying anything much at all.

Reply:
@gtully can't you instead make the test deterministic? like keeping consumers and senders on a certain condition that's required for the test?


I had this issue in other tests... where I needed to verify the bindings were in place before asserting.. or the consumers were ready to receive with credits.. etc...